FROM: Paul Elschot <paul.elsc...@xs4all.nl>
SUBJECT: Re: ArrayIndexOutOfBoundsException on BooleanScorer.score()
DATE: 6 May 2005

On Thursday 05 May 2005 21:45, Matt Magoffin wrote:
> The exception does come on the heels of an update to the index by a
> different thread than the one the search runs in. These log statements
> show the operations going on just prior to the exception:
> 
> May-05 12:36:05 DEBUG - Indexing Lead 1024
> May-05 12:36:06 TRACE - CON Closing IndexWriter
> org.apache.lucene.index.IndexWriter@4b35ef
> May-05 12:36:06 DEBUG - Parsing Lucene query string ...
> May-05 12:36:06 DEBUG - Searching for leads with Lucene query: ...
> May-05 12:36:06 TRACE - CON Thread[IndexQueueThread,5,main] releasing
> WRITE lock for index [LEAD]
> May-05 12:36:06 TRACE - CON Thread[main,5,main] got READ lock for index
> [LEAD]
> May-05 12:36:06 TRACE - CON Cached new IndexReader
> org.apache.lucene.index.SegmentReader@15c80a4
> May-05 12:36:06 TRACE - CON Created new IndexSearcher
> org.apache.lucene.search.IndexSearcher@1c1f2
> May-05 12:36:06 ERROR - Lucene exception during search on [LEAD]
> java.lang.ArrayIndexOutOfBoundsException: 4
>  ...
> May-05 12:36:06 TRACE - CON Thread[main,5,main] releasing READ lock for
> index [LEAD]
> 
> The search begins after the other thread's index update is complete,
> however (as shown by the "releasing WRITE lock" and "got READ lock"
> messages). Also a new IndexReader is created after the write completes in
> this situation.
> 
> Any ideas are greatly appreciated.

The size of the coordFactors array depends on the number of non prohibited
clauses in the query. Could you indicate which query you are using?

And in case you find a way to reproduce this in a  test case, could
you file a bug report in bugzilla? 

Regards,
Paul Elschot.


> 
> -- m@
> 
> 
> 
> > Hello, I'm having a tough time trying to get to the root of an exception I
> > see sometimes on my Lucene 1.4.3 index. The exception is:
> >
> > java.lang.ArrayIndexOutOfBoundsException: 4
> >  at org.apache.lucene.search.BooleanScorer.score(BooleanScorer.java:126)
> >  at org.apache.lucene.search.Scorer.score(Scorer.java:38)
> >  at org.apache.lucene.search.IndexSearcher.search(IndexSearcher.java:92)
> >  at org.apache.lucene.search.Hits.getMoreDocs(Hits.java:64)
> >  at org.apache.lucene.search.Hits.<init>(Hits.java:43)
> >  at org.apache.lucene.search.Searcher.search(Searcher.java:33)
> >  at org.apache.lucene.search.Searcher.search(Searcher.java:27)
> >  ...
> >
> > and occurs when current.coord is set to 4 but coordFactors is of length 4.
> >  The maxCoord is value is 4 at this time, too. I can't get this to happen
> > every time I run the unit test that this error occurs in... which is very
> > frustrating, sorry.
> >
> > I was wondering if anyone had an idea of what might cause this exception
> > (I'm assuming something is wrong on my use of Lucene, but I can't find
> > what).
> >
> >
> > ---------------------------------------------------------------------
> > To unsubscribe, e-mail: java-user-unsubscribe@lucene.apache.org
> > For additional commands, e-mail: java-user-help@lucene.apache.org
> >
> >
> 
> 
> ---------------------------------------------------------------------
> To unsubscribe, e-mail: java-user-unsubscribe@lucene.apache.org
> For additional commands, e-mail: java-user-help@lucene.apache.org
> 
> 
> 


---------------------------------------------------------------------
To unsubscribe, e-mail: java-user-unsubscribe@lucene.apache.org
For additional commands, e-mail: java-user-help@lucene.apache.org


